### PR TITLE
Resolved Issue with Running 'go' Command on Windows in Paths with Spaces

### PR DESCRIPTION
**Pull request by noyshabtay**
- [x] All [tests](https://ci.appveyor.com/project/jfrog-ecosystem/build-info) passed. If this feature is not already covered by the tests, I added new tests.
-----

### CODE REVIEW

File: build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/GoDriver.java [hunk 1]
```diff
@@ -31,12 +32,40 @@ public class GoDriver implements Serializable {
...
+        // If executablePath ends with "/go" - remove it
+        if (StringUtils.endsWith(executablePath, "go")) {
+            executablePath = StringUtils.substring(executablePath, 0, executablePath.length() - 3);
+        }
+
+        if (env.containsKey("Path")) {
+            env.put("Path", executablePath + File.pathSeparator + env.get("Path"));
+        } else {
+            env.put("Path", executablePath);
+        }
```
Comment (Or-Geva) [inline build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/GoDriver.java:63]: Let's add a comment here.

File: build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/GoDriver.java [hunk 2]
```diff
@@ -31,12 +32,40 @@ public class GoDriver implements Serializable {
...
+    /**
+     * Create a CommandExecutor with the given executable path and environment variables.
+     * Handle a bug in windows, where the go executable path was mistakenly considered as two command arguments
+     * in cases where the executable path contains spaces (for example: "C:\Program Files\Go\bin\go.exe").
+     *
+     * @param executablePath Go executable path
+     * @param env            Environment variables map
+     * @return CommandExecutor
+     */
+    private static CommandExecutor generateCommandExecutor(String executablePath, Map<String, String> env) {
```
Comment (Or-Geva) [inline build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/GoDriver.java:49]: We can include tests for this function that cover all possible cases. Each test can provide an explanation of what it aims to check and why.

**Review by Or-Geva — CHANGES_REQUESTED** [review]

File: build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/GoDriver.java [hunk 3]
```diff
@@ -29,13 +31,55 @@ public class GoDriver implements Serializable {
     private final CommandExecutor commandExecutor;
     private final File workingDirectory;
     private final Log logger;
+    private static final String defaultExecutablePath = "go";
```
Comment (yahavi) [inline build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/GoDriver.java:34]:
```suggestion
    private static final String DEFAULT_EXECUTABLE_NAME = "go";
```

Let's also put it above, near the other private static final variables.

File: build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/GoDriver.java [hunk 4]
```diff
@@ -29,13 +31,55 @@ public class GoDriver implements Serializable {
...
+     *
+     * @param executablePath Go executable path
+     * @param env            Environment variables map
+     * @return CommandExecutor
+     */
+    private static CommandExecutor generateCommandExecutor(String executablePath, Map<String, String> env) {
+        if (!SystemUtils.IS_OS_WINDOWS || StringUtils.isBlank(executablePath) || StringUtils.equals(executablePath, defaultExecutablePath) || env == null) {
+            return new CommandExecutor(StringUtils.defaultIfEmpty(executablePath, defaultExecutablePath), env);
+        }
+        // Handling Windows case:
```
Comment (yahavi) [inline build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/GoDriver.java:77]: But Windows cases can be in the above "if" too. Can we rearrange the code? Specifically, I'd simplify the "if" above.

File: build-info-extractor-go/src/test/java/org/jfrog/build/extractor/go/GoDriverTest.java [hunk 5]
```diff
@@ -114,4 +117,21 @@ public void testGoGet() throws IOException {
             FileUtils.deleteDirectory(projectDir);
         }
     }
+
+    @Test
+    public void testGoDriverWindowsInit() throws IOException {
+        if (!SystemUtils.IS_OS_WINDOWS) {
+            throw new SkipException("Skipping Windows test");
```
Comment (yahavi) [inline build-info-extractor-go/src/test/java/org/jfrog/build/extractor/go/GoDriverTest.java:124]: On the contrary - you skip non-windows tests

File: build-info-extractor-go/src/test/java/org/jfrog/build/extractor/go/GoDriverTest.java [hunk 6]
```diff
@@ -114,4 +117,21 @@ public void testGoGet() throws IOException {
...
+        }
+        File projectDir = Files.createTempDirectory("").toFile();
+        try {
+            Map<String, String> systemEnv = System.getenv();
+            String executablePath = "C:\\Program Files\\Go\\bin\\go";
+            Map<String, String> executorEnv = GoDriver.generateWindowsEnv(executablePath, systemEnv);
+            assertTrue(executorEnv.get("Path").startsWith("C:\\Program Files\\Go\\bin"));
+            new GoDriver(executablePath, systemEnv, projectDir, new NullLog());
```
Comment (yahavi) [inline build-info-extractor-go/src/test/java/org/jfrog/build/extractor/go/GoDriverTest.java:132]: What does this line do?

**Review by yahavi — CHANGES_REQUESTED** [review]

**Pull request merged**